pkg/operator/operator.go

```diff
@@ -157,14 +160,60 @@ func New(logger logr.Logger, clientConfig *rest.Config, registry prometheus.Regi
 	if err != nil {
 		return nil, errors.Wrap(err, "invalid port")
 	}
-	mgr, err := ctrl.NewManager(clientConfig, manager.Options{
+	manager, err := ctrl.NewManager(clientConfig, manager.Options{
 		Scheme: sc,
 		Host:   host,
 		Port:   port,
 		// Don't run a metrics server with the manager. Metrics are being served
 		// explicitly in the main routine.
 		MetricsBindAddress: "0",
-		CertDir:            certDir,
+		// Manage cluster-wide and namespace resources at the same time.
+		NewCache: cache.NewCacheFunc(func(config *rest.Config, options cache.Options) (cache.Cache, error) {
+			return cache.New(clientConfig, cache.Options{
+				Scheme: options.Scheme,
+				// The presence of metadata.namespace has special handling internally causing the
+				// cache's watch-list to only watch that namespace.
```

Comment:
do you have a link to the docs on this?

Reply:
It's an optimization in the code: https://github.com/kubernetes-sigs/controller-runtime/blob/release-0.12/pkg/cache/internal/informers_map.go#L294

That line looks for the presence of a metadata.namespace field in the selectors. If it exists (or the root manager specified a namespace, which we do not), then the watch only applies for that namespace.